File: models/bloodbank/BloodDonationCamp.js
// models/BloodDonationCamp.js
import mongoose from "mongoose";

const BloodDonationCampSchema = new mongoose.Schema(
  {
    hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
    branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

    campName: String,
    location: String,
    startDate: Date,
    endDate: Date,

    organizedBy: String,
    doctorInCharge: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
  },
  { timestamps: true }
);

export default mongoose.model("BloodDonationCamp", BloodDonationCampSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/BloodDonationCamp.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "blood_donation_camps")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BloodDonationCamp extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @Column(name = "camp_name")
    private String campName;
    private String location;
    @Column(name = "start_date")
    private LocalDateTime startDate;
    @Column(name = "end_date")
    private LocalDateTime endDate;

    @Column(name = "organized_by")
    private String organizedBy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctor_in_charge")
    private User doctorInCharge;
}
